docs/site/Relations.md

```diff
@@ -56,3 +59,39 @@ application.
 
 To generate a `HasMany` or `BelongsTo` relation through the CLI, see
 [Relation generator](Relation-generator.md).
+
+## Limitations
```

Comment:
The reason why I put limitations in each relation is because I found it weird to talk about navigational properties, related models, etc. before introducing them. New users might get lost. So I am a bit reluctant to put them in the top page.  

However,  I do agree with you that it makes it hard find the information. It is hard to find the balance. I don't have strong opinions on it. If others think it's fine I am okay with the change.